### ISSUE TITLE
space-before-function-parentheses false positive

### ISSUE THREAD

**Issue by feross** [eslint/eslint#1920]
eslint incorrectly reports an error for this code:

```js
var k = ''
var a = {
  set b (val) {
    k = val
  }
}
a.b = 'abc'
console.log(k)
```

Line 3: Missing space before function parentheses.

With this config:

```json
{
  "space-before-function-parentheses": [2, "always"]
}
```

**Comment by nzakas** [comment]
We probably just forgot to include getters/setters in the check.

**Comment by btmills** [comment]
Working on this.

**Comment by lo1tuma** [comment]
This could be also a problem for generator functions.

**Comment by btmills** [comment]
@lo1tuma Seems to me space-before-function-parentheses should ignore anonymous generator functions and let generator-star handle that case?
